### PR TITLE
Feature/data binding

### PR DESCRIPTION
**Pull request by m-the-hoff**
## Overview

Initial version of data binding and list virtualization framework.  See included READMEs for more information

## Changes

N/A

## Verification

- Proper use of best practices for MRTK and C# language conventions and features
- Directory organization of files
- Design weaknesses / flaws / bugs
- Documentation accuracy and completeness
- Ease of use
- Missing critical functionality
- Any other useful feedback

### CODE REVIEW

File: Assets/MRTK/SDK/Features/DataBinding/Scripts/DataConsumers/DataConsumerImageTextureFromUrl.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+            // UnityEngine.Debug.Log("SendWebRequest: " + uri);
+            for (int retries = 0; !doneWithSuccess && retries < kMaxRetries; retries++)
+            {
+                using (UnityWebRequest webRequest = UnityWebRequestTexture.GetTexture(uri))
+                {
+                    yield return webRequest.SendWebRequest();
+
+                    if (webRequest.isHttpError || webRequest.isNetworkError)
+                    {
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/DataBinding/Scripts/DataConsumers/DataConsumerImageTextureFromUrl.cs:130]: `webRequest.isHttpError` and `webRequest.isNetworkError` are obsolete starting in Unity 2020.2

https://docs.unity3d.com/2020.2/Documentation/ScriptReference/Networking.UnityWebRequest-isNetworkError.html

```
#if UNITY_2020_2_OR_NEWER
                    if (webRequest.result == UnityWebRequest.Result.ProtocolError || webRequest.result == UnityWebRequest.Result.ConnectionError)
                    {
#else
                    if (webRequest.isHttpError || webRequest.isNetworkError)
                    {
#endif
```

File: Assets/MRTK/SDK/Features/DataBinding/Scripts/DataSources/DataSourceGOJson.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        public IEnumerator StartJsonRequest(string uri, RequestSuccessDelegate successDelegate = null, RequestFailureDelegate failureDelegate = null, string requestId = null)
+        {
+            using (UnityWebRequest webRequest = UnityWebRequest.Get(uri))
+            {
+                // Request and wait for the desired page.
+                yield return webRequest.SendWebRequest();
+
+                if (webRequest.isHttpError || webRequest.isNetworkError)
+                {
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/DataBinding/Scripts/DataSources/DataSourceGOJson.cs:53]: Same comment as above

File: Assets/MRTK/SDK/Features/DataBinding/Scripts/Interfaces/IDataObjectPool.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    /// the frequency and duration of garbage collections. It can
+    /// also reduce the instantiation time of a prefab when populating
+    /// large lists.
+    /// </summary>
+    /// 
+    public interface IDataObjectPool
+    {
+
+        void SetMaximumPoolSize(int maxSize, bool resizeNow);
+
```
Comment (Zee2) [inline Assets/MRTK/SDK/Features/DataBinding/Scripts/Interfaces/IDataObjectPool.cs:22]: More whitespace; many other files have a lot of extra whitespace too, but I don't want to spam comments. Reducing unnecessary whitespace is great for readability!

**Review by Zee2 — COMMENTED** [review]
Most of these are small nits, aside from the comments about `?.`. Also, most of these files have quite a bit of extra whitespace; from prior experience from my own PRs, we tend to try and trim as much unnecessary whitespace as possible to improve readability/compactness/better use of IDE real estate!

File: Assets/MRTK/SDK/Features/DataBinding/Scripts/DataSources/DataSourceGOJson.cs (hunk 2 above)
Comment (saed219) [reply Assets/MRTK/SDK/Features/DataBinding/Scripts/DataSources/DataSourceGOJson.cs:60]: Egypt

**Comment by SimonDarksideJ** [comment]
Guessing that one of these branches needs synchronising with the current state of development as there are too many "dirty" changes included to review.  PR's should only include those changes the PR is intending to make.

**Comment by polar-kev** [comment]
There has been significant improvements to this feature in a separate repo so I'm going to close this PR. We will bring this feature to the community soon.
